app/routes/goal_routes.py

```diff
@@ -1 +1,83 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response, jsonify
+from ..db import db
+from app.models.goal import Goal
+from app.models.task import Task
+from .route_utilities import validate_model, create_model
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    goal_data, status_code = create_model(Goal, request_body)
+    return jsonify({"goal": goal_data}), status_code
+
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal)
+
+    title_param = request.args.get("title")
+    if title_param:
+        query = query.where(Goal.title.ilike(f"%{title_param}%"))
+
+    goals = db.session.scalars(query)
+
+    return [goal.to_dict() for goal in goals]
+
+
+@bp.get("/<id>")
+def get_one_goal(id):
+    goal = validate_model(Goal, id)
+    return jsonify({"goal": goal.to_dict()}), 200
```

Comment:
```suggestion
    return {"goal": goal.to_dict()}
```

Per my comment above, we should remove `jsonify` since we don't need it. 

Also, anywhere you return status code 200 can be removed since that's the default status code Flask will return.